core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -171,7 +182,11 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
       while (records.hasNext()) {
         final Product2<K, V> record = records.next();
         final K key = record._1();
-        partitionWriters[partitioner.getPartition(key)].write(key, record._2());
+        final int partitionId = partitioner.getPartition(key);
+        partitionWriters[partitionId].write(key, record._2());
+        if (rowBasedChecksums.length > 0) {
```

Comment:
We should consistently use either `null` to indicate checksum is disabled, or `length == 0`